src/devices/bus/mononcol/slot.cpp

```diff
@@ -43,9 +43,15 @@ void device_mononcol_cart_interface::rom_alloc(u32 size, int width, endianness_t
 
 	std::string fulltag(tag);
 	fulltag.append(MONONCOL_ROM_REGION_TAG);
+
 	device().logerror("Allocating %u byte ROM region with tag '%s' (width %d)\n", size, fulltag, width);
 	m_rom = device().machine().memory().region_alloc(fulltag, size, width, endian)->base();
 	m_rom_size = size;
+
+	if (!(m_rom_size && !(m_rom_size & (m_rom_size - 1))))
+	{
+		fatalerror("m_rom_size is NOT a power of 2");
+	}
```

Comment:
You’re better off detecting the error in `call_load`, calling `seterror`, and returning `image_init_result::FAIL`.  Throwing `emu_fatalerror` isn’t reported to the user in a very friendly way.

Reply:
I don't mind

Reply:
This whole member function is completely unnecessary – the slot can pass in the region allocated by the software list loader when loading from the software list, or it can allocate memory for loading from a loose file in `call_load`.  Most of the cartridge devices shouldn’t be allocating memory for ROM at all.

Reply:
I'm ok with it, here's one by me, it basically does the same thing:
https://github.com/mamedev/mame/blob/master/src/devices/bus/odyssey2/rally.cpp#L43

Reply:
Yes, but we’re trying to make the user experience better as we go, not spread existing problems to new drivers.

Reply:
I'd be highly surprised if anybody tries running the Monon driver with anything other than the software list ROMs anyway, it's not like there are thousands of bad size ROMs around for it.